Make it compile again after AVStream::codec deprecation

hirak99 opened the pull request:
This is a proposed set of changes to accommodate removal of AVStream::codec in ffmpeg 4.4.

- Converted AVCodecContext to AVCodecParameters wherever possible. This seems like a straight forward change.

Also -
- Populated the context and used it for rc_max_rate in dlnameta.c.
- Removed av_register_all() calls since it is no longer needed.

hirak99 committed:
Make it compile again after AVStream::codec deprecation

AVStream::codec is no longer in ffmpeg libraries, which prevents the code
to compile without these changes.

JuanKman94 commented on dlnameta.c:
```diff
@@ -374,15 +380,15 @@ get_dlna_metadata_video_ctx(struct AVFormatContext *ctx, int audio_stream, int v
-	struct AVCodecContext *ac = NULL;
-	struct AVCodecContext *vc = NULL;
+	struct AVCodecParameters *ac = NULL;
+	struct AVCodecParameters *vc = NULL;
```
When compiling this, I get a warning. `vc` is being passed to [lav_get_interlaced](https://github.com/JuanKman94/readymedia-transcode/blob/transcode/libav.h#L158). I tried replacing that parameter to `AVCodecParameters` but then I get a bunch of other warnings as [AVCodecParameters](https://libav.org/documentation/doxygen/master/structAVCodecParameters.html) does not have the [time_base](https://libav.org/documentation/doxygen/master/structAVCodecContext.html#ab7bfeb9fa5840aac090e2b0bd0ef7589) property used in [libav.h#163](https://github.com/JuanKman94/readymedia-transcode/blob/transcode/libav.h#L161)

What's weird is that the program did compile 🤔  Will it cause a runtime error? I'm no expert. Have you streamed a lot with this build? If so, then I guess it's okay

hirak99 replied:
Interesting. I compiled it in Arch, also tested that it installs with AUR when I changed the source to my fork.

However, I have not streamed with this yet.

I see that warning too. I believe it's worth keeping the compile clean if it was clean earlier. Let me take a look.

Also, thanks for checking :)

hirak99 replied:
Argh... I realize this needs more changes.

* ~I am assigning ac multiple times here https://github.com/hirak99/readymedia-transcode/blob/transcode/dlnameta.c#L234, but freeing it up only once.~ False alarm, it was being handled correctly. We are assigning it atmost once. Nonetheless, added a comment to the variable name.
* As I look at it in git, I realized I'm using spaces but the code uses tabs.

Let me fix all these and check that it streams.

hirak99 replied:
It's working, streamed my old videos successfully.
My TV doesn't support audio streaming, so I couldn't test that.
And another weird thing, after I disconnect a connection, the report at localhost:8200 sometimes shows "-1 connections currently open". Possibly an edge condition from before, and not introduced by this change?

hirak99 replied:
And compiles without that warning anymore.

JuanKman94 replied:
I've been using this branch for more than a week now, no hiccups :+1: I'm going to merge :smile: 

BTW, I haven't played too much with the decoding scripts but I'm going to tweak them to see if I can stream some MKV videos to an old samsung smart tv. I'll check you branch from time to time to see if you've changed the transcode scripts as well.

Thanks for the contribution!

JuanKman94 commented on dlnameta.c:
```diff
@@ -164,7 +164,7 @@ get_dlna_metadata_image(int fd)
 	m = get_dlna_metadata_image_res(format, width, height);
 	free(format);
-	
+
```
Didn't realize there were so much white space lines 👌

JuanKman94 commented on inotify.c:
```diff
@@ -666,8 +666,7 @@ start_inotify()
 	if (setpriority(PRIO_PROCESS, 0, 19) == -1)
 		DPRINTF(E_WARN, L_INOTIFY,  "Failed to reduce inotify thread priority\n");
 	sqlite3_release_memory(1<<31);
-	av_register_all();
```
Putting this issue here as reference. Removing this call is expected and does not affect behavior

https://github.com/leandromoreira/ffmpeg-libav-tutorial/issues/29

hirak99 committed:
Fixed call to lav_get_interlaced

Also -
- Fixed some formatting
- Renamed some variables to reflect the changes

hirak99 committed:
Silence the warning on unexpected socket close

Unexpected socket close is handled properly, this should be debug.
This reduces spamp to systemd logs.

hirak99 committed:
Handled the case where inotify add_watch fails for a known path
